tests/utils/test_notifications.py

```diff
@@ -1012,3 +1013,99 @@ def _generate_run_result(
         run_example["status"]["error"] = error
         run_example["status"]["state"] = state
     return run_example
+
+
+class TestMailNotification:
+    DEFAULT_PARAMS = {
+        "server_host": "smtp.gmail.com",
+        "server_port": 587,
+        "sender_address": "sender@example.com",
+        "username": "user",
+        "password": "pass",
+        "email_addresses": "a@example.com",
+        "use_tls": True,
+    }
+
+    @pytest.mark.parametrize(
+        "params, should_raise",
+        [
+            (
+                {
+                    "server_host": "smtp.gmail.com",
+                    "server_port": 587,
+                    "sender_address": "sender@example.com",
+                    "username": "user",
+                    "password": "pass",
+                    "email_addresses": "a@example.com",
+                },
+                False,
+            ),
+            (
+                {
+                    "server_host": "smtp.gmail.com",
+                    "server_port": 587,
+                    "sender_address": "sender@example.com",
+                    "username": "user",
+                    "password": "pass",
+                    "email_addresses": ["a@example.com", "b@example.com"],
+                },
+                False,
+            ),
+            (
+                {
+                    "server_port": 587,
+                    "sender_address": "sender@example.com",
+                    "username": "user",
+                    "password": "pass",
+                    "email_addresses": "a@example.com",
+                },
+                True,
+            ),
+        ],
+    )
+    def test_validate_mail_params(self, params, should_raise):
+        try:
+            mlrun.utils.notifications.notification.MailNotification.validate_params(
+                params
+            )
+        except ValueError:
+            assert should_raise
+        else:
+            assert not should_raise
```

Comment:
In a lot of mlrun tests we have the concept putting in the parametrize an `"expectation"`, that can have an empty context manager or a `pytest.raises(ValueError)`.

And then you can run:
```
with expectation:
    mlrun.utils.notifications.notification.MailNotification.validate_params(
        params
    )
```